util/util_test.go

```diff
@@ -792,3 +792,45 @@ func TestIsSupportedVersionSkew(t *testing.T) {
 		})
 	}
 }
+
+func TestRemoveOwnerRef(t *testing.T) {
+	g := NewWithT(t)
+	ownerRefs := []metav1.OwnerReference{
+		{
+			APIVersion: "dazzlings.info/v1",
+			Kind:       "Twilight",
+			Name:       "m4g1c",
+		},
+	}
+
+	tests := []struct {
+		name        string
+		toBeRemoved metav1.OwnerReference
+		wasRemoved  bool
+	}{
+		{
+			name: "owner reference present",
+			toBeRemoved: metav1.OwnerReference{
+				APIVersion: "dazzlings.info/v1",
+				Kind:       "Twilight",
+				Name:       "m4g1c",
+			},
+			wasRemoved: true,
+		},
+		{
+			name: "owner reference not present",
+			toBeRemoved: metav1.OwnerReference{
+				APIVersion: "dazzlings.info/v1",
+				Kind:       "Twilight",
+				Name:       "abcdef",
+			},
+			wasRemoved: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			newOwnerRefs := RemoveOwnerRef(ownerRefs, tt.toBeRemoved)
+			g.Expect(len(ownerRefs) != len(newOwnerRefs)).To(Equal(tt.wasRemoved))
```

Comment:
What about using `HasOwnerRef` to check if `toBeRemoved` is gone?

Reply:
Makes the test cleaner, good point